rtree.go

```diff
@@ -416,35 +416,31 @@ func (tree *Rtree) condenseTree(n *node) {
 
 // Searching
 
-// SearchIntersectBB returns all objects that intersect the specified rectangle.
-//
+// SearchIntersect returns all objects that intersect the specified rectangle.
 // Implemented per Section 3.1 of "R-trees: A Dynamic Index Structure for
 // Spatial Searching" by A. Guttman, Proceedings of ACM SIGMOD, p. 47-57, 1984.
-func (tree *Rtree) SearchIntersect(bb *Rect) []Spatial {
-	return tree.searchIntersect(-1, tree.root, bb)
-}
-
-// SearchIntersectWithLimit is similar to SearchIntersect, but returns
-// immediately when the first k results are found. A negative k behaves exactly
-// like SearchIntersect and returns all the results.
-func (tree *Rtree) SearchIntersectWithLimit(k int, bb *Rect) []Spatial {
-	return tree.searchIntersect(k, tree.root, bb)
+func (tree *Rtree) SearchIntersect(bb *Rect, filters ...Filter) []Spatial {
```

Comment:
Adding a new function which accepts filters, to which the existing function delegates, would avoid breaking existing clients who aren't interested in filtering. This seems to match your intention in adding the LimitFilter to this lib (as opposed to keeping it in client libraries), but it still requires client refactoring if we don't keep the current function signature. What are your thoughts?